docs/dev/background-information/sql/upgrades_and_migrator_design.mdx

```diff
@@ -0,0 +1,357 @@
+containerized as if it was another application which allows it to be run
+as an initContainer (in Kubernetes deployments) or as a dependent
+service (for Docker Compose deployments) to ensure that the necessary
+migrations are applied before startup of the application proper.
+Check out [RFC 469](https://docs.google.com/document/d/1_wqvNbEGhMzu45CHZNrLQYys7gl0AtVpLluGp4cbQKk/edit#heading=h.ddeuyk4t99yx) to learn more.
+
+### Important Commands
+
+The most important migrator commands are `up` and `upgrade`, with a notable mention to `drift`:
+
+-   **Up** : The default command of `migrator`, `up` ensures that
+    for a given version of the migrator, every migration defined at 
+    that build has been successfully applied in the connected database, 
+    this is specifically important to ensure *patch version migrations are run*. 
+    ***This must be run with the target version migrator image/build**.* 
+    If migrator and frontend pods are deployed in version lockstep, `up` 
+    ensures that ALL migrations required by the frontend will be successfully applied prior to boot.
+    This is a syntactic sugar over a more internal `upto` command.
+
+-   **Upgrade**: `upgrade` runs all migrations defined between two minor versions
+    of Sourcegraph and requires that other services which may access the
+    database are brought down. Before running, the database is checked
+    for and schema drifts in order to prevent a failure while attempting
+    a migration. `upgrade` relies on `stitched-migration-graph.json`.
+
+-   **Drift**: This command pulls runs a diff between the current database
+    schema and an expected definition packaged in migrator during the
+    release process. Many migrator operations run this check before 
+    proceeding to ensure the database is in the expected state.
+
+In general, the `up` command can be thought of as a **standard** upgrade
+(rolling upgrades with no downtime) while the upgrade command is what
+enables **multiversion** upgrades. In part, `up` was designed to maintain
+our previous upgrade policy and is thus run as an initContainer (or
+initContainer-like mechanism) of the frontend, i.e. between two versions
+of Sourcegraph, the antecedent sourcegraph services will continue to
+work after the consequent version's migrations have been applied.
+
+## Current Startup Dependency
+
+
+<img src="https://storage.googleapis.com/sourcegraph-assets/Docs/migrator-startup.png" alt="Migrator Startup Dependency" className="screenshot"/>
+
+All of our deployment types currently utilize migrator during startup.
+The frontend service won't start until migrator has been run with the
+default up command. The frontend service will also validate the expected
+schema (and OOB migration progress), and die on startup if this
+validation pass fails. This ensures that the expected migrations for the
+version in question have been run.
+
+In docker-compose (*see diagram)*, this is accomplished via a chain of
```

Comment:
Which diagram?

Reply:
This is one of the images thats loaded in, there are only two diagrams so its pretty clear when viewing a live page